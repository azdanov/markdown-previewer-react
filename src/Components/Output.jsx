// @flow

import React from 'react';
import marked from 'marked';

marked.setOptions({
  renderer: new marked.Renderer(),
  gfm: true,
  tables: true,
  breaks: true,
  pedantic: false,
  sanitize: true,
  smartLists: true,
  smartypants: false,
});

const Output = (props: { input: string }) => {
  const markdown = marked(props.input);
  return (
    <section
      className="output"
      // eslint-disable-next-line react/no-danger
      dangerouslySetInnerHTML={{ __html: markdown }}
    />
  );
};

export default Output;
