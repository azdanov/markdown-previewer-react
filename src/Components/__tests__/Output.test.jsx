// @flow

import React from 'react';

import { shallow, render } from 'enzyme';
import Output from '../Output';

test('Output renders correctly', () => {
  const component = shallow(<Output input={''} />);
  expect(component).toMatchSnapshot();
});

test('Output renders correct markup', () => {
  const input = 'Foo';
  const markdown = `# ${input}`;

  const component = render(<Output input={markdown} />);

  expect(component.text()).toContain(input);
  expect(component.find('h1').length).toBe(1);
});
