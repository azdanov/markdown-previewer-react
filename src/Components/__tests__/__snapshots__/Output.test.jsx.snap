// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Output renders correctly 1`] = `
<div
  className="output"
  dangerouslySetInnerHTML={
    Object {
      "__html": "",
    }
  }
/>
`;
